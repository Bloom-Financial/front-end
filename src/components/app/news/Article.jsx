import React from 'react';
import PropTypes from 'prop-types';
import style from './article.css';

const Article = ({ title, url, description, img }) => (
  <a href={url}>
    <figure className={style.articleContainer}>
      <h3 className={style.articleTitle}>{title}</h3>
      <img className={style.articleImg} src={img} alt={title} />
      <p className={style.articleDesc}>{description}</p>
    </figure>
  </a>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  img: PropTypes.string
};

export default Article;
