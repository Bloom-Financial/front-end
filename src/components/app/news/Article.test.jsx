import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';


describe('Test the Article component', () => {
  it('tests the Article component', () => {
    const { asFragment } = render(<Article/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
