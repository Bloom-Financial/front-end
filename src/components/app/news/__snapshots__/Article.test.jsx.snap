// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Article component tests the Article component 1`] = `
<DocumentFragment>
  <figure
    class="articleContainer"
  >
    <a>
      <h3
        class="articleTitle"
      />
       
    </a>
    <img
      class="articleImg"
    />
    <p
      class="articleDesc"
    />
  </figure>
</DocumentFragment>
`;
